Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM python:3.9-buster as base
+# Perform common operations, dependency installation etc...
+RUN apt-get update && apt-get install -y \
+    curl
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+# Adding poetry to PATH
+ENV PATH="/root/.poetry/bin:$PATH"
+
+RUN mkdir /app/
+COPY . /app/
+
+# Defaults
+WORKDIR /app/
+
+#RUN poetry install
+RUN poetry config virtualenvs.create false --local && poetry install --no-dev
```

Comment:
Not worth worrying about, but you haven't specified any dev dependencies in your `pyproject.toml`, so specifying `--no-dev` here doesn't change anything.